# Análise Exploratória dos Dados (EDA)

In [2]:
#Importando bibliotecas
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

import warnings
warnings.filterwarnings('ignore')

In [3]:
#Importar dataframe de treino e de teste
data_treino = pd.read_csv('titanic/train.csv')
data_teste = pd.read_csv('titanic/test.csv')

In [4]:
#Traduzir colunas
data_treino.columns = ['IdPassageiro', 'Sobreviveu',
                       'Classe', 'Nome', 'Sexo', 'Idade',
                       'IrmaosConjuge','PaisFilhos', 'Bilhete',
                       'Tarifa', 'Cabine', 'Embarque']

In [5]:
data_teste.columns = ['IdPassageiro', 'Classe', 'Nome',
                      'Sexo', 'Idade','IrmaosConjuge',
                      'PaisFilhos', 'Bilhete', 'Tarifa',
                      'Cabine', 'Embarque']

In [6]:
#Definir o IdPassageiro como índice
data_treino.set_index('IdPassageiro', inplace=True)
data_teste.set_index('IdPassageiro', inplace=True)

In [7]:
data_treino.shape, data_teste.shape

((891, 11), (418, 10))

In [8]:
#Concatenar dataframes
data = pd.concat([data_treino, data_teste],sort=True)

In [9]:
data['Sexo'] = data['Sexo'].map({'male': 'homem',
                                 'female': 'mulher'})

In [10]:
data.head()

Bilhete Cabine  Classe Embarque  Idade  IrmaosConjuge  \
IdPassageiro                                                                   
1                    A/5 21171    NaN       3        S   22.0              1   
2                     PC 17599    C85       1        C   38.0              1   
3             STON/O2. 3101282    NaN       3        S   26.0              0   
4                       113803   C123       1        S   35.0              1   
5                       373450    NaN       3        S   35.0              0   

                                                           Nome  PaisFilhos  \
IdPassageiro                                                                  
1                                       Braund, Mr. Owen Harris           0   
2             Cumings, Mrs. John Bradley (Florence Briggs Th...           0   
3                                        Heikkinen, Miss. Laina           0   
4                  Futrelle, Mrs. Jacques Heath (Lily May Peel)           0   
5                                      Allen, Mr. William Henry           0   

                Sexo  Sobreviveu   Tarifa  
IdPassageiro                               
1              homem         0.0   7.2500  
2             mulher         1.0  71.2833  
3             mulher         1.0   7.9250  
4             mulher         1.0  53.1000  
5              homem         0.0   8.0500

## Resumo da Análise Exploratória
### Parâmetros Categóricos
* Nominais
 * Sexo
 * Embarque
 * Sobreviveu
 
* Ordinais
 * Classe
 

### Parâmetros Númericos
* Contínuos
 * Idade
 * Tarifa
 
* Discretos
 * IrmaosConjuge
 * PaisFilhos
 
### Parâmetros Alfanuméricos
* Bilhete

# Engenharia de Recursos

In [11]:
data.isnull().sum()

Bilhete             0
Cabine           1014
Classe              0
Embarque            2
Idade             263
IrmaosConjuge       0
Nome                0
PaisFilhos          0
Sexo                0
Sobreviveu        418
Tarifa              1
dtype: int64

In [19]:
#Preencher dados faltando na coluna Embarque com a moda
moda_embarque = data['Embarque'].mode()[0]
data['Embarque'].fillna(moda_embarque, inplace=True)

#Preencher Tarifa faltando com a média
media_tarifa = data['Tarifa'].mean()
data['Tarifa'].fillna(media_tarifa, inplace=True)

In [21]:
#Remover coluna Cabine
data.drop('Cabine', axis=1, inplace=True)

In [23]:
#Informações gerais(Dados Numéricos)
data.describe()

Classe        Idade  IrmaosConjuge   PaisFilhos  Sobreviveu  \
count  1309.000000  1046.000000    1309.000000  1309.000000  891.000000   
mean      2.294882    29.881138       0.498854     0.385027    0.383838   
std       0.837836    14.413493       1.041658     0.865560    0.486592   
min       1.000000     0.170000       0.000000     0.000000    0.000000   
25%       2.000000    21.000000       0.000000     0.000000    0.000000   
50%       3.000000    28.000000       0.000000     0.000000    0.000000   
75%       3.000000    39.000000       1.000000     0.000000    1.000000   
max       3.000000    80.000000       8.000000     9.000000    1.000000   

            Tarifa  
count  1309.000000  
mean     33.295479  
std      51.738879  
min       0.000000  
25%       7.895800  
50%      14.454200  
75%      31.275000  
max     512.329200

In [24]:
#Informações gerais(Dados Categóricos)
data.describe(include=['O'])

Bilhete Embarque              Nome   Sexo
count       1309     1309              1309   1309
unique       929        3              1307      2
top     CA. 2343        S  Kelly, Mr. James  homem
freq          11      916                 2    843

In [27]:
data_treino['Sobreviveu'].value_counts()[1]

342

In [29]:
print('Tamanho da amostra: ',
      len(data))
print('Proporção total da amostra em relação à população: ',
      len(data_treino)/2224)
print('Proporção de Sobreviventes(População): ',
      1-1502/2224)
print('Proporção de Sobreviventes(Amostra): ',
     data_treino['Sobreviveu'].value_counts()[1]/len(data_treino))

Tamanho da amostra:  1309
Proporção total da amostra em relação à população:  0.4006294964028777
Proporção de Sobreviventes(População):  0.32464028776978415
Proporção de Sobreviventes(Amostra):  0.3838383838383838


In [31]:
#Obter taxa de sobrevivência por classe
data[['Classe', 'Sobreviveu']].groupby(['Classe']).mean().sort_values(
    by='Sobreviveu', ascending=False)

Sobreviveu
Classe            
1         0.629630
2         0.472826
3         0.242363

In [32]:
#Obter taxa de sobrevivência por Sexo
data[['Sexo', 'Sobreviveu']].groupby(['Sexo']).mean().sort_values(
    by='Sobreviveu', ascending=False)

Sobreviveu
Sexo              
mulher    0.742038
homem     0.188908

In [33]:
#Obter taxa de sobrevivência por IrmaosConjuge
data[['IrmaosConjuge', 'Sobreviveu']].groupby(['IrmaosConjuge']).mean()\
    .sort_values(by='Sobreviveu', ascending=False)

Sobreviveu
IrmaosConjuge            
1                0.535885
2                0.464286
0                0.345395
3                0.250000
4                0.166667
5                0.000000
8                0.000000

In [34]:
#Obter taxa de sobrevivência por PaisFilhos
data[['PaisFilhos', 'Sobreviveu']].groupby(['PaisFilhos']).mean()\
    .sort_values(by='Sobreviveu', ascending=False)

Sobreviveu
PaisFilhos            
3             0.600000
1             0.550847
2             0.500000
0             0.343658
5             0.200000
4             0.000000
6             0.000000
9                  NaN

## Resumo da Engenharia de Recursos
### Informações Faltando:
* Cabine(1014):
    * Coluna foi removida por conter muitos dados faltando
* Embarque(2):
    * Embarque é uma quantidade irrisória se comparado com a  amostra. Por este motivo foi preenchida com a moda
* Idade(263):
    * Idade será estimada através da regressão linear
* Sobreviveu(418):
    * A sobrevivência do passageiro será estimada através da regressão logistica
* Tarifa(1):
    * A Tarifa será preenchida com a média da tarifa cobrada dos passageiros

### Informações da população x amostra
- Tamanho da amostra:  1309
- Tamanho da população: 2224
- Proporção total da amostra em relação à população:  0.4006294964028777
- Proporção de Sobreviventes(População):  0.32464028776978415
- Proporção de Sobreviventes(Amostra):  0.3838383838383838

### Relações de Sobrevivência
- Passageiros da primeira classe estão mais propensos a sobreviver, seguido da segunda classe e com menores chances de sobrevivência a terceira classe
- Passageiros do sexo femino estão mais propensas a sobreviver.